ipatests/test_integration/test_epn.py

```diff
@@ -59,6 +62,79 @@ def postconf(host, option):
     host.run_command(r"postconf -e '%s'" % option)
 
 
+def generate_user(host, uid, **kwargs):
+    user_defaults = {
+        'objectClass': [
+            'ipaobject',
+            'person',
+            'top',
+            'ipasshuser',
+            'inetorgperson',
+            'organizationalperson',
+            'krbticketpolicyaux',
+            'krbprincipalaux',
+            'inetuser',
+            'posixaccount',
+            'ipaSshGroupOfPubKeys',
+            'mepOriginEntry'],
+        'ipaUniqueID': ['autogenerate'],
+        'loginShell': ['/bin/zsh'],
+        'uidNumber': ['-1'],
+        'gidNumber': ['-1'],
+    }
+
+    user = dict(user_defaults)
+    user['dn'] = 'uid={uid},cn=users,cn=accounts,{suffix}'.format(
+        uid=uid,
+        suffix=host.hostname,
+    )
+    user['uid'] = [uid]
+    user['displayName'] = ['{0} {0}'.format(uid)]
+    user['initials'] = ['{}{}'.format(uid[1], uid[-1])]
+    user['gecos'] = user['displayName']
+    user['sn'] = [uid]
+    user['homeDirectory'] = ['/other-home/{}'.format(uid)]
+    user['mail'] = ['{uid}@{domain}'.format(
+        uid=uid, domain=host.domain)]
+    user['krbPrincipalName'] = ['{uid}@{realm}'.format(
+        uid=uid, realm=host.realm)]
+    user['krbCanonicalName'] = ['{uid}@{realm}'.format(
+        uid=uid, realm=host.realm)]
+    user['givenName'] = [uid]
+    user['cn'] = ['{0} {0}'.format(uid)]
+
+    if "krbPasswordExpiration" in kwargs.keys():
+        user['krbPasswordExpiration'] = kwargs['krbPasswordExpiration']
+
+    if "userPassword" in kwargs.keys():
+        user['userPassword'] = kwargs['userPassword']
+
+    if "Mail2" in kwargs.keys():
+        user['Mail2'] = kwargs['Mail2']
+
+    if "Mail3" in kwargs.keys():
+        user['Mail3'] = kwargs['Mail3']
```

Comment:
Mail2 and Mail3 are not valid attributes. Are you trying to make it multi-valued?

Reply:
I am trying to add multiple emails.
Dummy `ldapsearch` showed, that multiple values are stored like this (this is supposedly `LDIF` format):

```
krbPrincipalName: jozkomrkvicka@never.gonna.give.you.up
mail: jozko@mrkvicka.com
mail: igor@hnizdo.com
mail: mehehe@foo.bar
```

Reply:
The value of mail should be a list and each unique address appended to that and the LDIFWriter will handle the rest.

Otherwise it will happily write out the Mail2 and Mail3 attributes but the the result will not be importable since those are not valid attribute names.